sim/hunter/talents.go

```diff
@@ -1,870 +1,823 @@
+}
+
+func (hunter *Hunter) applyInvigoration() {
+	if hunter.Talents.Invigoration == 0 || hunter.Pet == nil {
+		return
+	}
+
+	focusMetrics := hunter.NewFocusMetrics(core.ActionID{SpellID: 53253})
+
+	hunter.Pet.RegisterAura(core.Aura{
+		Label:    "Invigoration",
+		Duration: core.NeverExpires,
+		OnReset: func(aura *core.Aura, sim *core.Simulation) {
+			aura.Activate(sim)
+		},
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if !spell.ProcMask.Matches(core.ProcMaskMeleeSpecial | core.ProcMaskSpellDamage) {
+				return
+			}
+
+			if !result.DidCrit() {
+				return
+			}
+
+			hunter.AddFocus(sim, 3*float64(hunter.Talents.Invigoration), focusMetrics)
+		},
+	})
+}
+
+func (hunter *Hunter) applyCobraStrikes() {
+	if hunter.Talents.CobraStrikes == 0 || hunter.Pet == nil {
+		return
+	}
+
+	actionID := core.ActionID{SpellID: 53260}
+	procChance := 0.05 * float64(hunter.Talents.CobraStrikes)
+
+	hunter.Pet.CobraStrikesAura = hunter.Pet.RegisterAura(core.Aura{
+		Label:     "Cobra Strikes",
+		ActionID:  actionID,
+		Duration:  time.Second * 10,
+		MaxStacks: 2,
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			hunter.Pet.focusDump.BonusCritRating += 100 * core.CritRatingPerCritChance
+			if hunter.Pet.specialAbility != nil {
+				hunter.Pet.specialAbility.BonusCritRating += 100 * core.CritRatingPerCritChance
+			}
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+			hunter.Pet.focusDump.BonusCritRating -= 100 * core.CritRatingPerCritChance
+			if hunter.Pet.specialAbility != nil {
+				hunter.Pet.specialAbility.BonusCritRating -= 100 * core.CritRatingPerCritChance
+			}
+		},
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if spell.ProcMask.Matches(core.ProcMaskMeleeSpecial | core.ProcMaskSpellDamage) {
+				aura.RemoveStack(sim)
+			}
+		},
+	})
+
+	hunter.RegisterAura(core.Aura{
+		Label:    "Cobra Strikes",
+		Duration: core.NeverExpires,
+		OnReset: func(aura *core.Aura, sim *core.Simulation) {
+			aura.Activate(sim)
+		},
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if spell != hunter.ArcaneShot { // Only arcane shot, but also can proc on non crits
+				return
+			}
+
+			if sim.RandomFloat("Cobra Strikes") < procChance {
+				hunter.Pet.CobraStrikesAura.Activate(sim)
+				hunter.Pet.CobraStrikesAura.SetStacks(sim, 2)
+			}
+		},
+	})
+}
+func (hunter *Hunter) applyTermination() {
+	if hunter.Talents.Termination == 0 {
+		return
+	}
+
+	actionID := core.ActionID{SpellID: 83490}
+
+	focusMetrics := hunter.NewFocusMetrics(actionID)
+	core.MakePermanent(hunter.RegisterAura(core.Aura{
+		Label: "Termination",
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if sim.IsExecutePhase25() &&
+				spell == hunter.SteadyShot ||
+				spell == hunter.CobraShot {
```

Comment:
no need for this formating when the line is not long enough to overflow the normal screen size